test/Tester/Forwarding/ShutdownSiloTests.cs

```diff
@@ -43,6 +48,76 @@ public override TestCluster CreateTestCluster()
             await promisesAfterShutdown;
         }
 
+        [Fact, TestCategory("Functional"), TestCategory("Forward")]
+        public async Task SiloGracefulShutdown_NoFailureOnGatewayShutdown()
+        {
+            const int numberOfGrains = 2 * NumberOfSilos;
+
+            var grains = new List<ICounterGrain>();
+            for (var i = 0; i < numberOfGrains; i++)
+            {
+                grains.Add(await GetCounterGrainOnPrimary());
+            }
+
+            var cancellationTokenSource = new CancellationTokenSource();
+            var promise = CallIncrement(grains, cancellationTokenSource.Token);
```

Comment:
Isn't it an issue that `CallIncrement` is called on the main thread instead of on a different one (using `Task.Run()` or something), so that the while loop would be running independently from the main thread?